### ISSUE TITLE
Shows "Estimating..." when fully charged

### ISSUE THREAD

**Issue by mzur** [mzur/gnome-shell-batime#1]
The label shows "Estimating..." although the battery is fully charged. The dropdown menu says "Fully Charged" so it should be possible to detect this state. Make the label disappear in this case.

**Comment by Mershl** [comment]
I can reproduce this issue on the latest version (3.36) on Wayland.

Steps to reproduce:
- Have a battery not fully charged.
- Plug it in.
- Once 100% is reached Estimating... will contiuously be shown.

**Comment by mzur** [comment]
This bug was supposedly fixed two years ago. I use GNOME Shell 3.36.3 myself and everything works as expected. Can you please prodvide more details on your system? Do you have any other extensions installed?

**Comment by Mershl** [comment]
I've ran a few tests and the "issue" I was facing is not exactly the same. The Estimating... is shown for several minutes after it is fully charged. Afterwards it's gone.

- Fedora 32
- Gnome Shell 3.36.3
- Wayland session
- gnome-shell-batime Version 5 (from extensions.gnome.org)

Steps to reproduce:

1. Have a battery not fully charged.
2. Plug it in.
3. Once 100% is reached Estimating... will be shown for about 2 to 3 minutes.

Taskbar will show Estimating... for ~3 minutes:
![image](https://user-images.githubusercontent.com/520251/89730945-9c68cf80-da43-11ea-8c0c-c7d50c2525d8.png)

After ~3 minutes:
![image](https://user-images.githubusercontent.com/520251/89731068-77c12780-da44-11ea-99c0-1be8d29b0ef5.png)

**Comment by mzur** [comment]
In my experience, this is normal behavior. This can't be fixed on the part of this extension anyway.

**Comment by Mershl** [comment]
Noticed this "issue" coming back on Gnome 41.3 and a Thinkpad T495.

"Estimating..." is shown for several hours. Unplugging, Plugging in does still results in permanent "Estimating..." after reaching 100%. Seems there is a bigger issue in Gnome itself.

**Comment by Mershl** [comment]
It might actually be an issue with the extension. Maybe an battery/power API behaviour changed or the Notebooks power limiter is misinterpreted?

![image](https://user-images.githubusercontent.com/520251/154813678-335595bc-2b3a-4eca-bd3b-6ba0d59b116d.png)

![image](https://user-images.githubusercontent.com/520251/154813700-a7e84b58-723c-45a2-85ed-b58d0cc12008.png)

(Gnome 41.3 and a Thinkpad T495)

**Comment by Mershl** [comment]
Disabling the charge limit avoids the < 100% "Not Charging" state and fixes the infinite "Estimating..." message of the extension as well. Is it possible for the extension to handle this combination?

**Comment by Mershl** [comment]
![image](https://user-images.githubusercontent.com/520251/155007429-16ad921c-1d54-48a3-ae48-684c18e2cc20.png)

is reported as `UPower.DeviceState.PENDING_CHARGE` (enum: 5), which currently maps to "Estimating...".

Can we treat `PENDING_CHARGE` state the same as FULLY_CHARGED? What do you think @mzur?

**Comment by mzur** [comment]
I'd like to keep this consistent with the native [power status](https://gitlab.gnome.org/GNOME/gnome-shell/-/blob/main/js/ui/status/power.js). Please have a look there, maybe this was changed in recent GNOME versions.

**Comment by Mershl** [comment]
Looking into it, it seems it has always been that way on GNOME. It's just that some hardware suppliers limit the charge in a "non-standard" way leading to the 98% charged + `PENDING_CHARGE` state I'm seeing.

```
        else if (this._proxy.State === UPower.DeviceState.PENDING_CHARGE)
            return _('Not Charging');
```

The native power status maps this state to "Not charging" state (as in a determined state, not the "not sure" -> "Estimating..." state) with `TimeToFull` of 0. In my opinion an Empty battery time (like `FULLY_CHARGED`) makes sense for this combination - as "Estimating..." misleads in that the state is temporary.

**Comment by mzur** [comment]
Thanks! Please feel free to send a pull request that changes the behavior of this extension to what GNOME is currently doing.

**Comment by mzur** [comment]
Resolved by #8.
